tests/at_functional_test/test/at_client_sync_test.dart

```diff
@@ -89,10 +90,212 @@ void main() {
+  });
+
+  test('a test to verify multiple updates of a key is synced to the cloud',
+      () async {
+    var atClient = atClientManager.atClient;
+    var atKey = AtKey()
+      ..key = 'key1'
+      ..namespace = 'wavi';
+    var value1 = 'value1.key1';
+    var value2 = 'value2.key1';
+    var putResult = await atClient.put(atKey, value1);
+    expect(putResult, true);
+    putResult = await atClient.put(atKey, value2);
+    expect(putResult, true);
+    // waiting for 10 seconds for sync to complete.
+    atClientManager.atClient.syncService.sync();
+    atClientManager.atClient.syncService.sync();
+    await Future.delayed(Duration(seconds: 10));
+    var llookupVerbBuilder = LLookupVerbBuilder()
+      ..atKey = 'key1.wavi'
+      ..sharedBy = atSign
+      ..operation = 'all';
+    var getResponse =
+        await atClient.getRemoteSecondary()!.executeVerb(llookupVerbBuilder);
+    var transformedValue =
+        await GetResponseTransformer(atClient).transform(Tuple<AtKey, String>()
+          ..one = atKey
+          ..two = (getResponse));
+    // Decrypting and verifying the value from the remote secondary
+    expect(transformedValue.value, value2);
+  });
+
+  test('Update from server for a key that exists in local secondary', () async {
+    var atClient = atClientManager.atClient;
+    var atKey = AtKey()
+      ..key = 'testkey'
+      ..namespace = 'wavi';
+    var value = 'localvalue';
+    var putResult = await atClient.put(atKey, value);
+    expect(putResult, true);
+    var updateVerbBuilder = UpdateVerbBuilder()
+      ..atKey = 'testkey'
+      ..sharedBy = atSign
+      ..isPublic = true
+      ..value = value;
+    var updateResponse =
+        await atClient.getRemoteSecondary()!.executeVerb(updateVerbBuilder);
+    expect(updateResponse.isNotEmpty, true);
+    // waiting for 10 seconds for sync to complete.
+    await Future.delayed(Duration(seconds: 10));
+    atClientManager.atClient.syncService.sync();
+    atClientManager.atClient.syncService.sync();
+    var llookupVerbBuilder = LLookupVerbBuilder()
+      ..atKey = 'testkey.wavi'
+      ..sharedBy = atSign
+      ..operation = 'all';
+    var getResponse =
+        await atClient.getRemoteSecondary()!.executeVerb(llookupVerbBuilder);
+    var transformedValue =
+        await GetResponseTransformer(atClient).transform(Tuple<AtKey, String>()
+          ..one = atKey
+          ..two = (getResponse));
+    // Decrypting and verifying the value from the remote secondary
+    expect(transformedValue.value, value);
+    expect(
+        transformedValue.metadata!.updatedAt!.isBefore(DateTime.now()), true);
+  });
+}
+
+void onDoneCallback(syncResult) {
+  print('sync result $syncResult');
+  //always assert that the sync is successful when this method is triggered
+  expect(syncResult.syncStatus, SyncStatus.success);
+  //when this method is triggered always switch state to indicate that sync has been successful
```

Comment:
there's no code which does what this comment says is happening ... ?

Reply:
Updated